tests/php/json-api/test-class.json-api-jetpack-endpoints.php

```diff
@@ -18,13 +43,105 @@ private function set_globals() {
 	}
 
 	public function setUp() {
+		global $blog_id;
+
 		if ( ! defined( 'WPCOM_JSON_API__BASE' ) ) {
 			define( 'WPCOM_JSON_API__BASE', 'public-api.wordpress.com/rest/v1' );
 		}
 
 		parent::setUp();
 
 		$this->set_globals();
+
+		// Initialize some missing stuff for the API.
+		WPCOM_JSON_API::init()->token_details = array( 'blog_id' => $blog_id );
+	}
+
+	/**
+	 * Tests allows_site_based_authentication method.
+	 *
+	 * @author fgiannar
+	 * @covers WPCOM_JSON_API_Endpoint allows_site_based_authentication
+	 * @group json-api
+	 * @dataProvider data_provider_test_allows_site_based_authentication
+	 *
+	 * @param bool $allow_jetpack_site_auth The endpoint's `allow_jetpack_site_auth` value.
+	 * @param bool $is_user_logged_in If a user is logged in.
+	 * @param bool $result The expected result.
+	 */
+	public function test_allows_site_based_authentication( $allow_jetpack_site_auth, $is_user_logged_in, $result ) {
+
+		$endpoint = new Jetpack_JSON_API_Dummy_Endpoint(
+			array(
+				'stat'                    => 'dummy',
+				'allow_jetpack_site_auth' => $allow_jetpack_site_auth,
+			)
+		);
+
+		if ( $is_user_logged_in ) {
+			wp_set_current_user( self::$admin_user_id );
+		}
+
+		$this->assertEquals( $result, $endpoint->allows_site_based_authentication() );
+	}
+
+	/**
+	 * Tests api accessibility on a private site.
+	 *
+	 * @author fgiannar
+	 * @covers WPCOM_JSON_API switch_to_blog_and_validate_user
+	 * @group json-api
+	 * @dataProvider data_provider_test_private_site_accessibility
+	 *
+	 * @param bool            $allow_jetpack_site_auth The endpoint's `allow_jetpack_site_auth` value.
+	 * @param bool            $use_blog_token If we should simulate a blog token for this test.
+	 * @param bool            $user_can_read If the current user has read capability. When a blog token is used this has no effect.
+	 * @param WP_Error|string $result The expected result.
+	 */
+	public function test_private_site_accessibility( $allow_jetpack_site_auth, $use_blog_token, $user_can_read, $result ) {
+		// Private site.
+		update_option( 'blog_public', '-1' );
```

Comment:
@leogermani I'd love to use WorDBless in these unit tests. Do you think it could be done easily?

Reply:
In this situation, you don't need it because the tests already run in a full WP environment with a DB :)